reference/pipeline/stages/index.md

```diff
@@ -13,10 +13,24 @@ sidebar:
 This article describes the currently-supported stages that you can add to your
 Spinnaker pipelines.
 
+Note that when you're creating a pipeline, you probably won't see every stage
+that's listed here: you'll only see the stages that Spinnaker supports on your
+provider.
+
 ## General
 
 ### Bake
-Bake an image in the specified region.
+Bake an image from the specified packages. _Baking_ is the process of
+creating a machine image. Spinnaker provides default
+[Packer templates](https://www.packer.io/intro/) and base machine images in
+order to get you started, but see the
+[bakery configuration guide](/setup/bakery/) if you want to customize your bake
+process.
+
+Note that Spinnaker looks for a previously-created image before baking a new
+one. By default, Spinnaker skips the  **Bake** stage if an equivalent image
```

Comment:
These semantics are a little tricky, and based on how names are generated for the image that will be baked. I think either @duftler or @ezimanyi can shed some insight

Reply:
Effectively we generate a unique key for each bake based on the parameters (versioned package, base os, etc.).  In principle this means that if either the packages have changed (because of a new build, etc.) or if the paramters of the bake stage have changed (ie, using a new base image) we should trigger a new bake.  The details are here: https://github.com/spinnaker/rosco/blob/12add382dcc58f9bbbf4cce37ff49d4744c58fd8/rosco-core/src/main/groovy/com/netflix/spinnaker/rosco/providers/CloudProviderBakeHandler.groovy#L78

Reply:
Thanks for the clarification :) I've updated the text to talk about the process that you just described.